gneiss/_formula.py

```diff
@@ -0,0 +1,347 @@
+    a default labeling will be applied (i.e. `y1`, `y2`, ...)
+
+    >>> tree = TreeNode.read(['(c, (b,a)Y2)Y1;'])
+
+    Once these 3 variables are defined, a regression can be performed.
+    These proportions will be converted to balances according to the
+    tree specified.  And the regression formula is specified to run
+    `temp` and `ph` against the proportions in a single model.
+
+    >>> res = ols('temp + ph', proportions, env_vars, tree)
+
+    From the summary results of the `ols` function, we can view the
+    pvalues according to how well each individual balance fitted in the
+    regression model.
+
+    >>> res.pvalues
+           Intercept            ph      temp
+    Y1  2.479592e-01  1.990984e-11  0.243161
+    Y2  6.089193e-10  5.052733e-01  0.279805
+
+    We can also view the balance coefficients estimated in the regression
+    model. These coefficients can also be viewed as proportions by passing
+    `project=True` as an argument in `res.coefficients()`.
+
+    >>> res.coefficients()
+        Intercept            ph      temp
+    Y1  -0.000499  9.999911e-01  0.000026
+    Y2   1.000035  2.865312e-07 -0.000002
+
+    The balance residuals from the model can be viewed as follows.  Again,
+    these residuals can be viewed as proportions by passing `project=True`
+    into `res.residuals()`
+
+    >>> res.residuals()
+                  Y1            Y2
+    s1 -4.121647e-06 -2.998793e-07
+    s2  6.226749e-07 -1.602904e-09
+    s3  1.111959e-05  9.028437e-07
+    s4 -7.620619e-06 -6.013615e-07
+    s5 -1.332268e-14 -2.375877e-14
+
+    The predicted balances can be obtained as follows.  Note that the predicted
+    proportions can also be obtained by passing `project=True` into
+    `res.predict()`
+
+    >>> res.predict()
+              Y1        Y2
+    s1  1.000009  0.999999
+    s2  2.000000  0.999999
+    s3  2.999991  0.999999
+    s4  3.999982  1.000000
+    s5  4.999999  0.999998
+
+    The overall model fit can be obtained as follows
+
+    >>> res.r2
+    0.99999999997996369
+
+    See Also
+    --------
+    statsmodels.regression.linear_model.OLS
+    """
+    table, metadata, tree = _intersect_of_table_metadata_tree(table,
+                                                              metadata,
+                                                              tree)
+    ilr_table, basis = _to_balances(table, tree)
+    data = pd.merge(ilr_table, metadata, left_index=True, right_index=True)
+
+    fits = []
+    for b in ilr_table.columns:
+        # mixed effects code is obtained here:
+        # http://stackoverflow.com/a/22439820/1167475
+        stats_formula = '%s ~ %s' % (b, formula)
+
+        mdf = smf.ols(stats_formula, data=data, **kwargs).fit()
+        fits.append(mdf)
+    return RegressionResults(fits, basis=basis,
+                             feature_names=table.columns)
+
+
+def mixedlm(formula, table, metadata, tree, groups, **kwargs):
```

Comment:
Yes and no.

The idea was to provide the same sort of interface to the user, so that they can easily swap methods in and out.
The only difference between `mixedlm` and `ols` from an API point of view is that `mixedlm` has an required argument `groups`.